src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,254 @@
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Audio/SoundFileWriterOpus.hpp>
+#include <SFML/System/Err.hpp>
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+// Anonymous namespace
+namespace
+{
+    // Make sure to write int into buffer little endian
+    void writeUint32(std::vector<unsigned char>& buffer, const sf::Uint32 value)
+    {
+        buffer.push_back(static_cast<unsigned char>(value & 0x000000FF));
+        buffer.push_back(static_cast<unsigned char>((value & 0x0000FF00) >> 8));
+        buffer.push_back(static_cast<unsigned char>((value & 0x00FF0000) >> 16));
+        buffer.push_back(static_cast<unsigned char>((value & 0xFF000000) >> 24));
+    }
+}
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::check(const std::string& filename)
+{
+    std::string extension = filename.substr(filename.find_last_of(".") + 1);
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+
+    return extension == "opus";
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::SoundFileWriterOpus() :
+m_channelCount (0),
+m_file         (),
+m_ogg          (),
+m_opus         (NULL),
+m_packageNumber(0)
+{
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::~SoundFileWriterOpus()
+{
+    close();
+}
+
+
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::open(const std::string& filename, unsigned int sampleRate, unsigned int channelCount)
+{
+    // Save the channel count
+    m_channelCount = channelCount;
+    m_sampleRate = sampleRate;
+
+    // Initialize the ogg/opus stream
+    if (ogg_stream_init(&m_ogg, std::rand()) == -1)
+    {
+        err() << "Stream init of ogg/opus file \"" << filename << "\" failed" << std::endl;
+        close();
+        return false;
+    }
+
+    int status = OPUS_OK;
```

Comment:
What I meant before is rather to initialize the status to something that's not OK. I don't know about `opus_encoder_create`'s semantics, but there's a chance that it does not fill the status flag for some errors.

My thought was simply:
"by default, loading is not successful, unless the loading function explicitly says so"
rather than
"loading is successful, unless the loading function says otherwise" :grinning: